summaries/search-based-survey.md

```diff
@@ -0,0 +1,335 @@
+Therefore, it is difficult to expect full coverage of these techniques for any non-trivial programs.
+
+###### Structure-Oriented Approaches
+Structure-Oriented Approaches take a *divide and conquer* approach to obtaining full coverage. In this approach, a separate search is done for each uncovered element.
+
+**Branch-Distance-Oriented Approaches**: 
+Branch-Distance-Oriented Approaches use branch predicates in combination with Random Search or Genetic Algorithms for the more difficult cases.
+A path is chosen and the relevant branch predicates are extracted. A Genetic Algorithm is then used to find input data that satisfies all the branch predicates at once.
+Since this technique requires rigid constraints, the chance of getting stuck in local optima is high.
+It would be better if more feedback could be provided via the objective function.
+This is where control oriented approaches come into play.
+
+**Control-Oriented Approaches**:
+In this technique, the objective function considers branching nodes that need to be executed to cover a desired structure.
+This technique needs the control dependence graph of the test subject to identify each of these branching nodes.
+The problem with the Control-Oriented Approach is that the objective function gives no guidance on how to change the flow of execution at control dependent nodes, since no distance information is exploited from branch predicates.
+
+**Combined Approaches**
+Combined Approaches make use of both branch distance and control information for the objective function.
+The technique suffers a bit from local optima, but this can be resolved by using a form of *approximation level*.
+The result looks quite a bit like the Control-Oriented approach, but by using branch distance calculations the objective function is greatly improved.
+
+#### Objective Functions for Different Structural Coverage Criteria
+Structural criteria are divided into four categories, where the general objective function is defined as: `approach_level + m_branch_dist`.
+These two variables depends on the coverage type in question.
+
+**Node-Oriented:**
+Aim to cover specific nodes of the control flow graph.
+
+**Path-Oriented:**
+Require the execution of specific paths through the control flow graph.
+
+**Node-Path-Oriented:**
+Include branch coverage and LCSAJ (linear code sequence and jump) coverage.
+
+**Node-Node-Oriented:**
+Aim to execute a certain sequence of nodes through the control flow graph, without the specification of a concrete path between each node.
+
+#### Control-Related Problems for Objective Functions
+A major problem is covering nested structures within loops, which require many iterations.
+Some approaches try to resolve this by saying that the branches that miss the target in iterations of the loop as if they were critical branches.
```

Comment:
"by saying that the branches that miss the target in iterations of the loop as if they were critical branches"
I would rephrase this to
"by considering branches that miss the target in iterations of the loop to be critical branches"